app/webapp/src/main/java/uk/ac/ebi/fg/annotare2/web/gwt/editor/client/view/experiment/design/DataAssignmentViewImpl.java

```diff
@@ -178,20 +177,29 @@ private String getDataFileColumnName(FileType type, int index) {
     private List<EnumWithHelpText> getAllowedColumnTypes() {
         List<EnumWithHelpText> types = new ArrayList<>();
         for (FileType type : FileType.values()) {
-            if(!((experimentType.isTwoColorMicroarray()) && (type.isRawMatix())))
-            {
-                if (experimentType.isSequencing()) {
-                    if (!type.isFGEM() || type.isProcessed() && (0 == countColumnsByType(type))) {
-                        types.add(type);
-                    }
-                } else if (0 == countColumnsByType(type)) {
+            if(!type.isProcessedMatrix()){
+                if (experimentType.isSequencing()){
                     types.add(type);
                 }
+                else if (experimentType.isTwoColorMicroarray()){
+                    if(!type.isRawMatix()){
```

Comment:
The code would be cleaner if you would extract this 2 conditions into a method.
After that extraction you can also put this and the following condition into 1 `if` statement.